internal/uvm/vpmem.go

```diff
@@ -152,3 +155,8 @@ func (uvm *UtilityVM) RemoveVPMEM(ctx context.Context, hostPath string) (err err
 func (uvm *UtilityVM) PMemMaxSizeBytes() uint64 {
 	return uvm.vpmemMaxSizeBytes
 }
+
+// ExceededVPMem returns true if the addition of a new vpmem device exceeds uvm limits on vpmem
+func (uvm *UtilityVM) ExceededVPMem(fileSize int64) bool {
```

Comment:
@katiewasnothere - This isnt thread safe. Do we need to lock here?

Reply:
You're right. I'll make a separate PR for this fix.